From 42c78622d25c8cd3c154cb12ae1e39ea3b1e954e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lu=C3=ADs=20Pereira?= <luis.artur.pereira@gmail.com>
Date: Mon, 2 Feb 2015 11:12:08 -0800
Subject: [PATCH 06/27] QIconLoader: mark a helper type as movable

QIconDirInfo is held in Qt containers, so reap the performance
benefit of a movable type.

Port of commit
https://qt.gitorious.org/qt/qtbase/commits/91f1b0b4e443c7c95f6a1654d9f6ac63ab778c77
---
 qiconfix/qiconloader_p.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/qiconfix/qiconloader_p.h b/qiconfix/qiconloader_p.h
index 77ab520..86776d4 100644
--- a/qiconfix/qiconloader_p.h
+++ b/qiconfix/qiconloader_p.h
@@ -62,6 +62,7 @@
 //#include <private/qicon_p.h>
 //#include <private/qfactoryloader_p.h>
 #include <QtCore/QHash>
+#include <QtCore/QTypeInfo>
 
 //QT_BEGIN_NAMESPACE
 
@@ -194,6 +195,10 @@ private:
 
 } // QtXdg
 
+// Note: class template specialization of 'QTypeInfo' must occur at
+//       global scope
+Q_DECLARE_TYPEINFO(QtXdg::QIconDirInfo, Q_MOVABLE_TYPE);
+
 //QT_END_NAMESPACE
 
 #endif // QT_NO_ICON
-- 
1.9.0

